### PR TITLE
2.x: move blocking operators into the base classes

### PR DESCRIPTION
**Pull request by akarnokd**
This PR removes the `BlockingFlowable` and `BlockingObservable` classes and moves the blocking operators into the respective reactive classes and names prefixed with `blocking`.

In addition, many of the mentions of `Flowable` has been replaced with `Observable` in `Observable`.

Related: #4366.

### CODE REVIEW

File: src/main/java/io/reactivex/Flowable.java [hunk 1]
```diff
@@ -5056,29 +5056,420 @@ private static void verifyPositive(long value, String paramName) {
...
+     * </dl>
+     *
+     * @param initialValue
+     *            the initial value that the {@link Iterable} sequence will yield if this
+     *            {@code BlockingObservable} has not yet emitted an item
+     * @return an {@link Iterable} that on each iteration returns the item that this {@code BlockingObservable}
+     *         has most recently emitted
+     * @see <a href="http://reactivex.io/documentation/operators/first.html">ReactiveX documentation: First</a>
+     */
+    public final Iterable<T> blockingMostRecent(T initialValue) {
```
Comment (JakeWharton) [inline src/main/java/io/reactivex/Flowable.java:5292]: What's the difference between "most recent" and "latest"?

File: src/main/java/io/reactivex/Flowable.java [hunk 2]
```diff
@@ -5056,29 +5056,420 @@ private static void verifyPositive(long value, String paramName) {
...
+     * <dl>
+     *  <dt><b>Backpressure:</b></dt>
+     *  <dd>The operator consumes the source {@code Flowable} in an unbounded manner
+     *  (i.e., no backpressure applied to it).</dd>
+     *  <dt><b>Scheduler:</b></dt>
+     *  <dd>{@code blockingLatest} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     *
+     * @return an Iterable that always returns the latest item emitted by this {@code BlockingObservable}
+     * @see <a href="http://reactivex.io/documentation/operators/first.html">ReactiveX documentation: First</a>
```
Comment (JakeWharton) [inline src/main/java/io/reactivex/Flowable.java:5266]: Wrong link?

File: src/main/java/io/reactivex/Flowable.java [hunk 3]
```diff
@@ -5056,29 +5056,420 @@ private static void verifyPositive(long value, String paramName) {
...
+     *  <dt><b>Scheduler:</b></dt>
+     *  <dd>{@code blockingMostRecent} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     *
+     * @param initialValue
+     *            the initial value that the {@link Iterable} sequence will yield if this
+     *            {@code BlockingObservable} has not yet emitted an item
+     * @return an {@link Iterable} that on each iteration returns the item that this {@code BlockingObservable}
+     *         has most recently emitted
+     * @see <a href="http://reactivex.io/documentation/operators/first.html">ReactiveX documentation: First</a>
```
Comment (JakeWharton) [inline src/main/java/io/reactivex/Flowable.java:5290]: Wrong link?

Comment (JakeWharton) [inline src/main/java/io/reactivex/Flowable.java:5290]: Nevermind. I see "latest" under there.

Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java:5290]: It's a copy of 1.x: https://github.com/ReactiveX/RxJava/blob/1.x/src/main/java/rx/observables/BlockingObservable.java#L293

**Comment by JakeWharton** [comment]
LGTM 👍

File: src/main/java/io/reactivex/Flowable.java (hunk 1 above)
Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java:5292]: Most recent keeps reemitting the current value while the push side just overwrites the current value. Latest takes the current value and blocks until one becomes available again; the push side just overwrites the current value.

**Pull request merged**